src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,571 @@
+        @param data address we are setting it to
+    */
+    function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositHub/no-file-during-shutdown");
+
+        if (what == "vow") vow = data;
+        else if (what == "end") end = EndLike(data);
+        else revert("DssDirectDepositHub/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    /**
+        @notice update tau value for D3M ilk.
+        @dev msg.sender must be authorized.
+        @param ilk  bytes32 of the D3M ilk to be updated
+        @param what bytes32("tau") or it will revert
+        @param data number of second to wait after caging a pool to write of debt
+    */
+    function file(bytes32 ilk, bytes32 what, uint256 data) external auth {
+        require(ilks[ilk].tic == 0, "DssDirectDepositHub/pool-not-live");
+
+        if (what == "tau" ) {
+            ilks[ilk].tau = data;
+        } else revert("DssDirectDepositHub/file-unrecognized-param");
+
+        emit File(ilk, what, data);
+    }
+
+    /**
+        @notice update plan or pool addresses for D3M ilk.
+        @dev msg.sender must be authorized.
+        @param ilk  bytes32 of the D3M ilk to be updated
+        @param what bytes32("pool"|"plan") or it will revert
+        @param data address we are setting it to
+    */
+    function file(bytes32 ilk, bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositHub/no-file-during-shutdown");
+        require(ilks[ilk].tic == 0, "DssDirectDepositHub/pool-not-live");
+
+        if (what == "pool") ilks[ilk].pool = D3MPoolLike(data);
+        else if (what == "plan") ilks[ilk].plan = D3MPlanLike(data);
+        else revert("DssDirectDepositHub/file-unrecognized-param");
+        emit File(ilk, what, data);
+    }
+
+    // --- Deposit controls ---
+    function _wind(bytes32 ilk, D3MPoolLike pool, uint256 amount) internal {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+        if (amount == 0) {
+            emit Wind(ilk, 0);
+            return;
+        }
+
+        require(int256(amount) >= 0, "DssDirectDepositHub/overflow");
+
+        vat.slip(ilk, address(pool), int256(amount));
+        vat.frob(ilk, address(pool), address(pool), address(this), int256(amount), int256(amount));
+        // normalized debt == erc20 DAI (Vat rate for this ilk fixed to 1 RAY)
+        daiJoin.exit(address(pool), amount);
+        pool.deposit(amount);
+
+        emit Wind(ilk, amount);
+    }
+
+    function _unwind(bytes32 ilk, D3MPoolLike pool, uint256 supplyReduction, uint256 availableAssets, Mode mode, uint256 assetBalance) internal {
```

Comment:
Since we have the pool as an argument, instead of passing `availableAssets` in I'd just read it within this function--its only use in `exec` is to be passed here.